inferno-core/test/Eval/Spec.hs

```diff
@@ -426,25 +426,27 @@ cachedGet =
     TestEnv {cache} <- liftImplEnvM $ ask
     pure $ VInt cache
 
-evalInMonadPrelude :: ModuleMap (ReaderT TestEnv IO) TestCustomValue
+-- Since this is a test, we build a prelude from scratch, instead of extending the
+-- builtin/core Inferno prelude from Inferno.Module.Prelude.
+-- To keep the rest of the code happy, we need to include a dummy Base module.
+evalInMonadPrelude :: Prelude (ReaderT TestEnv IO) TestCustomValue
 evalInMonadPrelude =
-  [infernoModules|
-module EvalInMonad
+  [builtinPreludeQuoter|
+module Base
+  zero : int := 0;
 
+module EvalInMonad
   cachedGet : () -> int := ###!cachedGet###;
 |]
 
 evalInMonadTest :: Spec
 evalInMonadTest = do
   let testEnv = TestEnv {cache = 4}
 
-  let modules =
-        Map.unionWith
-          (error "Duplicate module name in builtinModules")
-          (Prelude.builtinModules @(ReaderT TestEnv IO) @TestCustomValue)
-          evalInMonadPrelude
```

Comment:
The nice thing about chaining prelude evaluation is that we don't have to combine the preludes in a hacky way like this anymore